modules/lightning_telemetry/src/Telemetry.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Drupal\lightning_telemetry;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Extension\ModuleExtensionList;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Telemetry service.
+ */
+class Telemetry {
+
+  /**
+   * Amplitude API URL.
+   */
+  const AMPLITUDE_API_URL = 'https://api.amplitude.com/httpapi';
+
+  /**
+   * Amplitude API key.
+   *
+   * This is not intended to be private. It is typically included in client
+   * side code. Fetching data requires an additional API secret.
+   */
+  const AMPLITUDE_API_KEY = 'f32aacddde42ad34f5a3078a621f37a9';
+
+  /**
+   * The module handler to invoke the alter hook with.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * The extension.list.module service.
+   *
+   * @var \Drupal\Core\Extension\ModuleExtensionList
+   */
+  protected $extensionListModule;
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $config_factory;
+
+  /**
+   * Constructs a telemetry object.
+   *
+   * @param \Drupal\Core\Extension\ModuleExtensionList $extension_list_module
+   *   The extension.list.module service.
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config.factory service.
+   */
+  public function __construct(ModuleExtensionList $extension_list_module, ClientInterface $http_client, ConfigFactoryInterface $config_factory) {
+    $this->extensionListModule = $extension_list_module;
+    $this->httpClient = $http_client;
+    $this->config_factory = $config_factory;
+  }
+
+  /**
+   * Sends an event to Amplitude.
+   *
+   * @param \Drupal\lightning_telemetry\Event $event
+   *   The Amplitude event.
+   *
+   * @return bool
+   *   TRUE if the request to Amplitude was successful.
+   */
+  protected function sendEvent(Event $event) {
+    $response = $this->httpClient->request('POST', self::AMPLITUDE_API_URL, [
+      'form_params' => [
+        'api_key' => self::AMPLITUDE_API_KEY,
+        'event' => $event->__toJson(),
```

Comment:
We should generally avoid calling magic methods directly. Presumably, using \Drupal\Component\Serialization\Json::encode() will automatically call the __toJson() method.

Reply:
This is not a magic method, and it doesn't look like \Drupal\Component\Serialization\Json::encode() will call it.